mon-pix/app/services/locale.js

```diff
@@ -19,36 +23,79 @@ export default class LocaleService extends Service {
   @service dayjs;
   @service metrics;
 
+  get supportedLocales() {
+    return SUPPORTED_LOCALES;
+  }
+
+  get currentLocale() {
+    return this.intl.primaryLocale;
+  }
+
+  get acceptLanguageHeader() {
+    if (this.currentDomain.isFranceDomain) return FRENCH_FRANCE_LOCALE;
+    return this.currentLocale;
+  }
+
+  get availableLanguagesForSwitcher() {
+    const FRENCH_LANGUAGE = 'fr';
+    const options = Object.entries(VISIBLE_LANGUAGES)
+      .filter(([_, config]) => config.languageSwitcherDisplayed)
+      .map(([key, config]) => ({ label: config.value, value: key }));
+
+    return options.sort((option) => (option.value === FRENCH_LANGUAGE ? -1 : 1));
+  }
+
   isSupportedLocale(locale) {
     try {
       const localeCanonicalName = Intl.getCanonicalLocales(locale)?.[0];
-      return SUPPORTED_LOCALES.some((supportedLocale) => localeCanonicalName == supportedLocale);
+      return this.supportedLocales.some((supportedLocale) => localeCanonicalName == supportedLocale);
     } catch {
       return false;
     }
   }
 
-  handleUnsupportedLanguage(language) {
+  setLocale(locale) {
+    this.metrics.context.locale = locale;
+    this.intl.setLocale(locale);
+    this.dayjs.setLocale(locale);
+  }
+
+  setUserLocale(currentUser = null, overridingLanguage = null) {
```

Comment:
Cette fonction gagnerait à être mieux nommée. En effet elle ne change pas la locale de l’utilisateur. Elle positionne la locale courante en fonction de priorités, le domaine, le language, la `lang` de cet utilisateur.

Voici une proposition de nommage sûrement imparfaite mais pour donner une idée pour aiguiller les réflexions : 

```suggestion
  setBestLocale({ language, currentUser }) {
```

Reply:
Renommé en `detectBestLocale({ language, currentUser })`